services/filesstore/s3_overrides.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package filesstore
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// customTransport is used to point the request to a different server.
+// This is helpful in situations where a different service is handling AWS S3 requests
+// from multiple Mattermost applications, and the Mattermost service itself does not
+// have any S3 credentials.
+type customTransport struct {
+	base   http.RoundTripper
+	host   string
+	scheme string
+	cli    http.Client
```

Comment:
Not a big fan of using `cli` for clients, it always reminds me CLI. I can be biased against it 😅 I'd use `client` instead. 0/5

Reply:
lol, yeah I have used `client` here https://github.com/mattermost/mattermost-server/pull/15948/files. Will change.